server/filestore.go

```diff
@@ -3126,17 +3082,25 @@ checkCache:
 func (mb *msgBlock) fetchMsg(seq uint64) (*fileStoredMsg, error) {
 	var sm *fileStoredMsg
 
+	mb.mu.Lock()
+	defer mb.mu.Unlock()
+
 	sm, err := mb.cacheLookup(seq)
 	if err == nil || (err != errNoCache && err != errPartialCache) {
 		return sm, err
 	}
 
 	// We have a cache miss here.
-	if err := mb.loadMsgs(); err != nil {
+	if err := mb.loadMsgsWithLock(); err != nil {
 		return nil, err
 	}
 
-	return mb.cacheLookup(seq)
+	sm, err = mb.cacheLookup(seq)
+	if err == errNoCache {
+		fmt.Printf("NO CACHE [%d] FOR %d!!\n\n", mb.index, seq)
```

Comment:
Remove this print, or if needed for debug, make it a log/debug statement.

Reply:
Thanks fixed.